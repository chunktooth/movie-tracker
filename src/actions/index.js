export const logInUser = (id, name) => ({
  type: 'LOGIN_USER',
  id,
  name
});

export const logOutUser = () => ({
  type: 'LOGOUT_USER'
});

export const loadMovies = (movies) => ({
  type: 'LOAD_MOVIES',
  movies
});