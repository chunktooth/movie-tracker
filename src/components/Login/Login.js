import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

export class Login extends Component {
	constructor() {
		super();
		this.state = {
			username: '',
			password: ''
		}
	}

	render() {

		return (
			<form onSubmit={(e) => {
				e.preventDefault() 
				// handleSubmit(this.state)
			}}>
				<label for="username">Username: </label>
				<input 
					type='text' 
					id='username' 
					value={this.state.username} 
					placeholder='username'
					onChange={(e) => this.setState({ username: e.target.value })}
				/>
				<label for="password">Password: </label>
				<input 
					type='password' 
					id='password' 
					value={this.state.password} 
					placeholder='password'
					onChange={(e) => this.setState({ password: e.target.value })}
				/>
				<NavLink to='/home'>Sign in</NavLink>
			</form>
		)
	}
}