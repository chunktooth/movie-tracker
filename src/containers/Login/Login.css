form {
  width: 45%;
  margin: auto;
}

input {
  background-color: transparent;
  border: 2px solid #8C8C8C;
  border-radius: 20px;
  outline: none;
  margin-bottom: 10px;
  padding: 10px 10px 10px 30px;
  font-size: 16px;
  color: #E5E4E4;
}

button.submit {
  background-color: #8C8C8C;
  border: none;
  outline: none;
  border-radius: 20px;
  margin-top: 16px;
  padding: 10px;
  font-size: 16px;
}

button.submit:hover {
  background-color: #545454;
}

section {
  position: relative;
  top: 20px;
  text-align: center;
  animation: shake 0.5s;
}