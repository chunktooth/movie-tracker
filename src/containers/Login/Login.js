import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { addNewUser } from '../../actions';

export class Login extends Component {
	constructor() {
		super();
		this.state = {
			username: '',
			password: ''
		}
	}

	submitUsername = (e) => {
		e.preventDefault()
		this.props.handleSubmit(this.state.username, this.state.password);
		const path = "/";
		this.props.history.push(path);
	}


	render() {
		return (
			<form onSubmit={this.submitUsername}>
				<label htmlFor="username">Username: </label>
				<input 
					type='text' 
					id='username' 
					value={this.state.username} 
					placeholder='username'
					onChange={(e) => this.setState({ username: e.target.value })}
				/>
				<label htmlFor="password">Password: </label>
				<input 
					type='password' 
					id='password' 
					value={this.state.password} 
					placeholder='password'
					onChange={(e) => this.setState({ password: e.target.value })}
				/>

				<button>
					submit
				</button>
			</form>
		)
	}
}

const mapStateToProps = (state) => {
  return {
    user: state.user
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    handleSubmit: (username, password) => {
      dispatch(addNewUser(username, password))
    }
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Login))
