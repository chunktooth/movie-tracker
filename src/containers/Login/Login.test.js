import React from 'react';
import { shallow } from 'enzyme';
import { Login } from './Login';
import * as actions from '../../actions/index';

describe('Login', () => {
	let wrapper, mockHandleSubmit, mockEvent;

	beforeEach(() => {
		mockHandleSubmit = jest.fn();
		mockEvent = {
			preventDefault: jest.fn()
		}
		wrapper = shallow(<Login 
			handleSubmit={mockHandleSubmit}/>);
	});

	it.skip('should match the snapshot', ()=> {
		expect(wrapper).toMatchSnapshot();
	});

	it('should call handleSubmit when submit the form', () => {
		const username = 'jared';
		const password = 'hotstuff81';
		wrapper.instance().submitUsername(mockEvent);
		expect(mockHandleSubmit).toHaveBeenCalled();
	});

});