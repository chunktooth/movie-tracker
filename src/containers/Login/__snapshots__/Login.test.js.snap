// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    handleSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="username"
>
          Username: 
</label>,
        <input
          id="username"
          onChange={[Function]}
          placeholder="username"
          type="text"
          value=""
/>,
        <label
          htmlFor="password"
>
          Password: 
</label>,
        <input
          id="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
/>,
        <button>
          submit
</button>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Username: ",
          "htmlFor": "username",
        },
        "ref": null,
        "rendered": "Username: ",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "username",
          "onChange": [Function],
          "placeholder": "username",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Password: ",
          "htmlFor": "password",
        },
        "ref": null,
        "rendered": "Password: ",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "password",
          "onChange": [Function],
          "placeholder": "password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "submit",
        },
        "ref": null,
        "rendered": "submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="username"
>
            Username: 
</label>,
          <input
            id="username"
            onChange={[Function]}
            placeholder="username"
            type="text"
            value=""
/>,
          <label
            htmlFor="password"
>
            Password: 
</label>,
          <input
            id="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
/>,
          <button>
            submit
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Username: ",
            "htmlFor": "username",
          },
          "ref": null,
          "rendered": "Username: ",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "username",
            "onChange": [Function],
            "placeholder": "username",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Password: ",
            "htmlFor": "password",
          },
          "ref": null,
          "rendered": "Password: ",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "submit",
          },
          "ref": null,
          "rendered": "submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
